src/main/java/duke/data/TaskList.java

```diff
@@ -0,0 +1,235 @@
+package duke.data;
+
+import java.io.FileNotFoundException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import duke.DukeException;
+import duke.task.DateTimeTask;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+public class TaskList {
+
+    /** ArrayList to store all tasks */
+    private ArrayList<Task> items;
+    /** Calendar object to store dates and times of timed tasks */
+    private Calendar calendar;
+
+    /**
+     * Returns a new TaskList object when there is no previously saved data.
+     * Item list and calendar initialized to be empty.
+     */
+    public TaskList() {
+        this.items = new ArrayList<>();
+        this.calendar = new Calendar();
+    }
+
+    /**
+     * Returns a new TaskList object when there is previously saved data.
+     * Converts saved data to Task objects and initializes item list and calendar with saved data.
+     *
+     * @param taskArrayList ArrayList of String arrays, each representing a task.
+     * @throws DukeException If data is corrupt or missing information.
+     * @throws FileNotFoundException If data file cannot be found.
+     */
+    public TaskList(ArrayList<String[]> taskArrayList) throws DukeException, FileNotFoundException {
+        this.items = new ArrayList<>();
+        this.calendar = new Calendar();
+        for (String[] taskElements : taskArrayList) {
+            Task t = null;
+            switch (taskElements[0]) {
+            case "T":
+                t = new Todo(taskElements[2]);
+                break;
+            case "E":
+                t = new Event(Arrays.copyOfRange(taskElements, 2, taskElements.length));
+                calendar.add((DateTimeTask) t);
+                break;
+            case "D":
+                t = new Deadline(Arrays.copyOfRange(taskElements, 2, taskElements.length));
+                calendar.add((DateTimeTask) t);
+                break;
+            default:
+                break;
+            }
```

Comment:
How about reading the coding standard on switch-cases once again, to make sure this part is correct?

Reply:
Sorry my bad, as we already have confirmed on telegram, your coding standard on switch case is correct and I made a mistake due to a bug in my IDE while opening your file! Sorry for the confusion!